plane/source/docs/guided-mode.rst

```diff
@@ -4,12 +4,53 @@
 GUIDED Mode
 ===========
 
-The GUIDED mode is used when you want the aircraft to fly to a specific
-point on the map without setting up a mission. Most ground control
-stations support a "click to fly to" feature where you can click a point
-on the map and the aircraft will fly to that location then loiter.
+GUIDED mode is used when you want the aircraft to fly to a specific
+point on the map, or in specific direction (heading) without setting up a mission. 
+
+GUIDED mode is also used when an external controller such as Lua on the AutoPilot, 
+or ROS2 or a Python system using pymavlink control the aircraft by sending MAVLink 
+"guide slew" commands.
 
 The other major use for GUIDED mode is in :ref:`geo-fencing <geofencing>`.
 When the geo-fence is breached the aircraft will enter GUIDED mode, and
 head to the predefined geo-fence return point, where it will loiter
 until the operator takes over.
+
+Location sub-mode
+-----------------
+Most ground control
+stations support a "click to fly to" feature where you can click a point
+on the map and the aircraft will fly to that location then loiter. If an 
+altitude is requested, the aircraft will try to achieve the target altitude as it flies,
+and will loiter down or up to the requested altitude once it arrives, if required. 
+
+.. image:: ../../../images/guided-altitude.png
+
+This sub-mode can also be requested via MAVLink using the DO_REPOSITION command.
+
+If the vehicle is flying in Guided "Location sub-mode" then GUIDED_CHANGE_ALTITUDE and GUIDED_CHANGE_SPEED 
+commands will be ignored.
+
+Heading/Altitude sub-mode
+-------------------------
+This mode is used when sending Guided Slew Commands.
+
+The guided mode heading feature isn't supported by Mission Planner or QGroundControl, but can be
+requested via MAVLink commands or using Lua. If the aircraft receives a GUIDED_CHANGE_HEADING
+command it will fly indefinitely towards the heading in degrees. 
+
+The target altitude and desired airspeed or
+groundspeed of the vehicle to fly at can be set using the GUIDED_CHANGE_ALTITUDE and GUIDED_CHANGE_SPEED
```

Comment:
You can't control the target ground speed. It just sets the min ground speed param. We don't really support setting ground speed in plane.

Reply:
I've basically changed it to say what you said.